test/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationServiceTest.scala

```diff
@@ -15,29 +16,34 @@ class OAuth2TokenValidationServiceTest extends BaseSpec with Matchers with Route
   val mockTokenValidationService = ValidationTestRegistry.tokenValidationService
   val realValidationService = ValidationServiceRegistry.tokenValidationService
 
-  "Validation service" should "reject dummy token after validation remote server" in {
+  "Validation service" must "reject dummy token after validation remote server" in {
     whenReady(mockTokenValidationService.validate(Some(wrongToken))) { either =>
-      either should be('left)
+      either must be('left)
     }
   }
 
-  "Validation service" should "accept correct token after validation remote server" in {
+  "Validation service" must "accept correct token after validation remote server" in {
     whenReady(mockTokenValidationService.validate(Some(correctToken))) { either =>
-      either should be('right)
+      either must be('right)
     }
   }
 
-  "Validation service" should "validate access token (jwt) locally with public key" in {
+  "Validation service" must "validate access token (jwt) locally with public key" in {
     whenReady(realValidationService.validate(Some(correctToken), JWTTokenMethod)) { either =>
-      either should be('right)
+      either must be('right)
     }
   }
 
-  "Validation service" should "validate reject wrong token (not jwt)" in {
+  "Validation service" must "reject wrong token (not jwt)" in {
     whenReady(realValidationService.validate(Some(wrongToken), JWTTokenMethod)) { either =>
-      either should be('left)
+      either must be('left)
     }
   }
 
-
+  "Validation service" must "validate access token (jwt) and make sure that JWT is valid at least for next month" in {
+    whenReady(realValidationService.validate(Some(correctToken), JWTTokenMethod)) { either =>
+      either must be('right)
+      either.right.get.exp.get must be > System.currentTimeMillis() / 1000 + (60 * 24 * 30)
```

Comment:
```
either.right.get.exp.get must be > System.currentTimeMillis() / 1000 + (60 * 24 * 30)
```
is testing not for 30 days, but for half a day.
if you wish to test for 30 days, use `> System.currentTimeMillis() / 1000 + (3600 * 24 * 30)` or add joda time and use the proper constants so one doesn't need to spend half a minute to make sure each of these lines of constructing the value of "month" is actually correct.